internal/publish/preflight_checks.go

```diff
@@ -30,17 +29,12 @@ func (p *defaultPublisher) checkConfiguration(client connect.APIClient) error {
 	}
 	log.Info("Publishing with credentials", "username", user.Username, "email", user.Email)
 
+	var existingContentID *types.ContentID
 	if p.Target != nil {
-		log.Debug("Target found while checking configuration")
-		previousType := p.Target.Type
-		currentType := p.Config.Type
-
-		if previousType != "" && previousType != config.ContentTypeUnknown && currentType != previousType {
-			log.Debug("Content type mismatch between previous and new target", "previous_type", previousType, "current_type", currentType)
-			return types.OperationError(op, errTypeChanged)
-		}
+		existingContentID = &p.Target.ID
 	}
-	err = client.CheckCapabilities(p.Dir, p.Config, log)
+
+	err = client.CheckCapabilities(p.Dir, p.Config, existingContentID, log)
```

Comment:
Having to use a pointer for the ContentID feels a bit off, WDYT about passing the `Target` to the `CheckCapabilities`? then, you would only need one check, the one you have here above moved over to `ChackCapabilities`.
e.g:
```go
err = client.CheckCapabilities(p.Dir, p.Config, p.Target, log)

...
// CheckCapabilities
func (c *ConnectClient) CheckCapabilities(base util.AbsolutePath, cfg *config.Config, target *deployment.Deployment, log logging.Logger) error {
    if target != nil {
        err := c.ValidateDeploymentTarget(target.ID, log)
        if err != nil {
            return err
        }
}
```

Reply:
@marcosnav It's a decent idea, but unfortunately, the inclusion of a `deployment` import brings us to a circular dependency that doesn't look quick to unwind. OK if we leave this for later?

>package github.com/posit-dev/publisher/cmd/publisher
        imports github.com/posit-dev/publisher/cmd/publisher/commands
        imports github.com/posit-dev/publisher/internal/clients/connect
        imports github.com/posit-dev/publisher/internal/deployment
        imports github.com/posit-dev/publisher/internal/inspect/dependencies/renv
        imports github.com/posit-dev/publisher/internal/bundles
        imports github.com/posit-dev/publisher/internal/clients/connect: import cycle not allowed

Reply:
Ah, lovely circular dependencies 😬 !

Yes, I'm okay with leaving this as is, hitting circular dependencies issues is food for thought, though. Very likely we are sharing similar responsibilities between different instances, and/or not using interfaces enough. Worth revisiting at some point.

Reply:
Totally agree, it is a very big warning sign. We'll revisit this no doubt and get it straightened out.